sentry-ruby/lib/sentry/event.rb

```diff
@@ -9,7 +9,10 @@
 require 'sentry/utils/custom_inspection'
 
 module Sentry
+  # This is an abstract class that defines the shared attributes of an event.
+  # Please don't use it directly. The user-facing classes are its child classes.
   class Event
+    TYPE = "event"
```

Comment:
@sl0thentr0py By giving all events a default `"event"` type, I think we can remove many fallback code that deals with `nil` type. I think the events can still be accepted without any issue. Let me know if that'll cause any problem?

Reply:
oh yea that's perfect, this always felt like a code smell